ui/src/app/applications/components/application-create-panel/application-create-panel.tsx

```diff
@@ -311,18 +309,48 @@ export const ApplicationCreatePanel = (props: {
                                                 )}
                                             </div>
                                         );
-
                                         const destinationPanel = () => (
                                             <div className='white-box'>
                                                 <p>DESTINATION</p>
-                                                <div className='argo-form-row'>
-                                                    <FormField
-                                                        formApi={api}
-                                                        label='Cluster'
-                                                        field='spec.destination.server'
-                                                        componentProps={{items: clusters}}
-                                                        component={AutocompleteField}
-                                                    />
+                                                <div className='row argo-form-row'>
+                                                    {(destFormat.toUpperCase() === 'URL' && (
```

Comment:
The `server` field should be removed when user selects 'Name' in dropdown. If user select cluster URL first and then select Name in dropdown then both fields have values:
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: ''
spec:
  destination:
    name: in-cluster
    namespace: ''
    server: 'https://kubernetes.default.svc'
```